### DISCUSSION TITLE
View source and Edit this page using 'master' instead of 'main'

### DISCUSSION THREAD

**Discussion by golfecholima** [squidfunk/mkdocs-material#7810] [Q&A] [answered]
Seems like the default setting is to `master` rather than `main` which is causing 404s on those buttons. 

For example, the generated URL is https://github.com/user/repo/raw/master/docs/index.md ... but should be https://github.com/user/repo/raw/main/docs/index.md because that's my primary branch.

Where can I change that?

**Comment by niclasheinz** [comment]
Can you provide more information on your question? Do you mean publishing the documentation on GitHub Pages? Are you talking about the ‘Edit’ button?

**Reply by golfecholima** [reply]
Yes, the site is published to GitHub pages. 

I'm talking about both the View Source and Edit Page icons at the top right of a docs page:
<img width="120" alt="Screenshot 2024-12-18 at 12 12 13 PM" src="https://github.com/user-attachments/assets/60406a63-7e9c-483f-877f-41bc66e2e923" />

**Reply by niclasheinz** [reply, marked as answer]
Thanks for the clarification.

You can simple add this to your `mkdocs.yml` configuration:

```
edit_uri: -/edit/main/docs/
```
then you get the right url points to branch `main` for both edit and view source

Link to documentation: https://squidfunk.github.io/mkdocs-material/setup/adding-a-git-repository/?h=edit+uri#code-actions

**Reply by golfecholima** [reply]
Ugh! I wasn't searching for the right thing in the docs. Sorry. Thank you!

**Reply by kamilkrzyskow** [reply]
Searching for the title in the Discussions leads to other threads on the same topic:
https://github.com/squidfunk/mkdocs-material/discussions?discussions_q=View+source+and+Edit+this+page+using+%27master%27+instead+of+%27main%27
- https://github.com/squidfunk/mkdocs-material/discussions/4055

So it's worth searching the Discussions as well ✌️

**Reply by golfecholima** [reply]
Yeah, I did that -- just read past the solution in #4055. 🙈
